src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -336,12 +336,13 @@ private static (ControlInfoJson.Item item, int index) CombineIRAndState(BlockNod
                     StyleName = state.StyleName,
                     ExtensionData = state.ExtensionData,
                     IsGroupControl = state.IsGroupControl,
-                    GroupedControlsKey = state.GroupedControlsKey
+                    GroupedControlsKey = state.GroupedControlsKey,
+                    AllowAccessToGlobals = (state.IsComponentDefinition ?? false) ? (templateState?.ComponentManifest?.AllowAccessToGlobals ?? true) : true,
```

Comment:
this default to true has broken our existing roundtrip tests, looks like most apps don't have this set at all.